packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -207,14 +207,31 @@ You can set functionPerRoute: false to prevent surpassing the limit.`
 
 				// Multiple entrypoint support
 				if (_entryPoints.size) {
-					for (const [route, entryFile] of _entryPoints) {
-						const func = basename(entryFile.toString()).replace(/\.mjs$/, '');
+					// TODO: Currently this function directly checks for a single route 
+					// injected in packages/astro/src/assets/internal.ts Ideally this
+					// should be done dynamically
+					const isInjectedRoute = (route: RouteData) =>
+						route.pathname === '/_image';
```

Comment:
I think we could change the name in `isImageRoute`, and remove the `TODO`. Here we want to specially allow-list our internal `_image` endpoint.

Reply:
Makes sense.

I wonder if there should be any comment left regarding the todo here:
https://github.com/withastro/astro/blob/main/packages/astro/src/assets/internal.ts#L14-L19, just for the reference.

Reply:
Maybe if `route.component` doesn't start with `src/pages/`, we can then fallback for `getInjectedFuncName`? That way we don't have to hardcode this.

Reply:
Way better

Reply:
Yup looks great! Maybe the functions can be renamed so it's clearer, but the idea and flow is good.